### PR TITLE
Support `Origin: null` in preflight requests

### LINKED ISSUE #43: Support `Origin: null` in preflight requests
To be consistent with actual requests (fixed in #31), preflight requests should also accept `Origin: null`.

### PR DESCRIPTION
**Pull request by jonfox**
It looks like preflight requests with Origin: null are rejected - in my case, an OpenAPI page generated in IntelliJ as file:/// to use for testing.

It seems that we accept actual requests with Origin: null since #32, is there any reason not to also do so for preflight requests? Perhaps this was the problem @ShuiShengYe had per the #32 comments.

Another option could be to add explicit config to allow Origin: null in this case (e.g. that could be enabled in test scope), although with `allowed-origins = "*"` we are already fairly permissive.

Fix #43

### CODE REVIEW

**Comment by lomigmegard** [comment]
Thanks for the PR!

It makes sense indeed, I don't think the W3C explicitly forbids it (https://www.w3.org/TR/cors/#resource-preflight-requests)

File: akka-http-cors/src/test/scala/ch/megard/akka/http/cors/CorsDirectivesSpec.scala [hunk 1]
```diff
@@ -68,6 +68,21 @@ class CorsDirectivesSpec extends WordSpec with Matchers with Directives with Sca
...
+        status shouldBe StatusCodes.OK
+        response.headers should contain theSameElementsAs Seq(
+          `Access-Control-Allow-Origin`.`null`,
+          `Access-Control-Allow-Methods`(settings.allowedMethods),
+          `Access-Control-Max-Age`(1800),
+          `Access-Control-Allow-Credentials`(true)
+        )
+      }
+    }
+
```
Comment (lomigmegard) [inline akka-http-cors/src/test/scala/ch/megard/akka/http/cors/CorsDirectivesSpec.scala]: Could you add a test checking that when `allowed-origins` is set to something else than `*`, the `Origin: null` should be rejected.

Comment (jonfox) [reply akka-http-cors/src/test/scala/ch/megard/akka/http/cors/CorsDirectivesSpec.scala]: Added
